src/solc_0.7/GameToken/GameToken.sol

```diff
@@ -0,0 +1,436 @@
+
+        if (editors.length != 0) {
+            for (uint256 i = 0; i < editors.length; i++) {
+                _gameEditors[gameId][editors[i]] = true;
+            }
+        }
+
+        if (assetIds.length != 0) {
+            if (assetIds.length == 1 && values[0] == 1) {
+                addSingleAsset(from, gameId, assetIds[0], uri);
+            } else {
+                addMultipleAssets(from, gameId, assetIds, values, uri);
+            }
+        }
+        setTokenURI(gameId, uri);
+        emit AssetsAdded(gameId, assetIds, values);
+        return gameId;
+    }
+
+    /// @notice Function to burn a GAME token
+    /// @param from The address of the one destroying the game
+    /// @param to The address to send all game assets to
+    /// @param gameId The id of the game to destroy
+    function destroyGame(
+        address from,
+        address to,
+        uint256 gameId
+    ) external override minterGuard() {
+        require(from == _ownerOf(gameId), "DESTROY_ACCESS_DENIED");
+        require(to != address(this), "DESTINATION_GAME_CONTRACT");
+        (uint256[] memory assets, uint256[] memory values) = getGameAssets(gameId);
+        removeMultipleAssets(gameId, assets, values, to, "");
+        assert(_gameData[gameId]._assets.length() == 0);
+        _burn(from, gameId);
+    }
+
+    function _burn(address from, uint256 gameId) private {
+        delete _gameData[gameId];
+        delete _metaData[gameId];
+        _creatorship[creatorOf(gameId)] = address(0);
+        // @review best practices re: transfer to zero address
+        _transferFrom(from, address(0), gameId);
+        emit Transfer(from, address(0), gameId);
+    }
+
+    /// @notice Function to get game editor status
+    /// @param gameId The id of the GAME token owned by owner
+    /// @param editor The address of the editor to set
+    /// @return isEditor Editor status of editor for given tokenId
+    function isGameEditor(uint256 gameId, address editor) external view override returns (bool isEditor) {
+        return _gameEditors[gameId][editor];
+    }
+
+    /// @notice Get the creator of the token type `id`.
+    /// @param id the id of the token to get the creator of.
+    /// @return the creator of the token type `id`.
+    function creatorOf(uint256 id) public view override returns (address) {
+        require(id != uint256(0), "GAME_NEVER_MINTED");
+        address originalCreator = address(id / CREATOR_OFFSET_MULTIPLIER);
+        address newCreator = _creatorship[originalCreator];
+        if (newCreator != address(0)) {
+            return newCreator;
+        }
+        return originalCreator;
+    }
+
+    /// @notice Function to get all assets and their quantities for a GAME
+    /// @param gameId The id of the GAME to get assets for
+    function getGameAssets(uint256 gameId) public view override returns (uint256[] memory, uint256[] memory) {
+        uint256 assetLength = _gameData[gameId]._assets.length();
```

Comment:
This function is useful in some case, but mostly off-chain and as such can be emulated via listing events
The approach taken by ERC1155 is to not have such list function.

The benefit is that it would make it more efficient to add assets as we just need a `mapping(gameID => mapping(assetId => amount)`

the disadvantage would be that we woudl need a backend to kepp track of which assets and amount a game token hold.

Reply:
I've removed this function. We will rely on the backend to enumerate the assets in a GAME moving forward.